### PR TITLE
Add support for Snowflake data stream APIs

### PR DESCRIPTION
**Pull request by rsulli-rai**
* add support cli for data streams APIs
* minor typo cleanup

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.18
 
 require (
 	github.com/pkg/errors v0.9.1
-	github.com/relationalai/rai-sdk-go v0.4.4-alpha
+	github.com/relationalai/rai-sdk-go v0.4.5-alpha
 	github.com/spf13/cobra v1.5.0
```
Comment (rsulli-rai) [inline go.mod:8]: don't merge until sdk version is available. 
must run go mod tidy/vendor

File: rai/cmds.go [hunk 2]
```diff
@@ -881,6 +881,65 @@ func listSnowflakeDatabaseLinks(cmd *cobra.Command, args []string) {
 	action.Exit(rsp, err)
 }
 
+//
+// Snowflake database links
```
Comment (wienleung) [inline rai/cmds.go:885]: nit: should be datastreams

File: rai/cmds.go [hunk 3]
```diff
@@ -782,9 +782,15 @@ func updateUser(cmd *cobra.Command, args []string) {
...
 //
 
+var ErrNoEngineForIntegration = errors.New("an integration must be assigned an engine")
+
 func createSnowflakeIntegration(cmd *cobra.Command, args []string) {
 	action := newAction(cmd)
 	name := args[0]
+	engine := action.getString("engine")
+	if engine == "" {
+		action.Exit(nil, ErrNoEngineForIntegration)
```
Comment (stania-rai) [inline rai/cmds.go:792]: can we use `.MarkFlagRequired` instead?

File: rai/cmds.go [hunk 4]
```diff
@@ -881,6 +887,72 @@ func listSnowflakeDatabaseLinks(cmd *cobra.Command, args []string) {
 	action.Exit(rsp, err)
 }
 
+//
+// Snowflake data streams
+//
+
+func createSnowflakeDatastream(cmd *cobra.Command, args []string) {
```
Comment (stania-rai) [inline rai/cmds.go:894]: nit: inconsistent spelling - SDK uses `DataStream`

File: rai/main.go [hunk 5]
```diff
@@ -367,6 +368,48 @@ func addCommands(root *cobra.Command) {
...
+		Use:   "create-snowflake-data-stream integration database-link objectName",
+		Short: "Create Snowflake data stream associated with an integration",
+		Args:  cobra.ExactArgs(3),
+		Run:   createSnowflakeDatastream}
+	cmd.Flags().String("role", "", "Snowflake role (default: SNOWSQL_ROLE env var)")
+	cmd.Flags().String("warehouse", "", "Snowflake warehouse (default: SNOWSQL_WAREHOUSE env var)")
+	cmd.Flags().String("username", "", "Snowflake username (default: SNOWSQL_USER env var)")
+	cmd.Flags().String("password", "", "Snowflake password (default: SNOWSQL_PWD env var)")
+	cmd.Flags().Bool("is-view", false, "Snowflake object is a view (default: false)")
+	cmd.Flags().String("rai-database", "", "RelationAI target database name")
```
Comment (stania-rai) [inline rai/main.go:382]: `RelationalAI`. Same typo 2 lines below

File: rai/main.go [hunk 6]
```diff
@@ -367,6 +368,48 @@ func addCommands(root *cobra.Command) {
...
+	cmd.MarkFlagRequired("rai-database")
+	cmd.Flags().String("relation", "", "RelationAI relation")
```
Comment (stania-rai) [inline rai/main.go:384]: `target relation` in the description is probably more descriptive

File: rai/cmds.go [hunk 7]
```diff
@@ -881,6 +884,72 @@ func listSnowflakeDatabaseLinks(cmd *cobra.Command, args []string) {
...
+		integration, dbLink, dataStream, role, &creds,
+	)
+	action.Exit(nil, err)
+}
+
+func getSnowflakeDataStream(cmd *cobra.Command, args []string) {
+	action := newAction(cmd)
+	integration := args[0]
+	dataStream := args[1]
+	dbLink := args[2]
```
Comment (stania-rai) [inline rai/cmds.go:938]: wrong order of args? should be `integration dbLink dataStream`

File: rai/cmds.go [hunk 8]
```diff
@@ -782,9 +782,12 @@ func updateUser(cmd *cobra.Command, args []string) {
 // Snowflake integrations
 //
 
+var ErrNoEngineForIntegration = errors.New("an integration must be assigned an engine")
```
Comment (stania-rai) [inline rai/cmds.go:785]: not used anymore..

**Review by stania-rai — APPROVED** [review]

**Pull request merged**